taca/utils/bioinfo_tab.py

```diff
@@ -0,0 +1,163 @@
+
+import socket
+import os
+import couchdb
+import glob
+import re
+import logging
+
+from csv import DictReader
+from taca.utils.config import CONFIG
+from flowcell_parser.classes import XTenSampleSheetParser
+
+logger = logging.getLogger(__name__)
+
+def setupServer(conf):
+    db_conf = conf['statusdb']
+    url="http://{0}:{1}@{2}:{3}".format(db_conf['username'], db_conf['password'], db_conf['url'], db_conf['port'])
+    return couchdb.Server(url)
+
+def merge(d1, d2):
+    """ Will merge dictionary d2 into dictionary d1.
+    On the case of finding the same key, the one in d1 will be used.
+    :param d1: Dictionary object
+    :param s2: Dictionary object
+    """
+    for key in d2:
+        if key in d1:
+            if isinstance(d1[key], dict) and isinstance(d2[key], dict):
+                merge(d1[key], d2[key])
+            elif d1[key] == d2[key]:
+                pass # same leaf value
+        else:
+            d1[key] = d2[key]
+    return d1
+
+
+def collect_runs():
+    rundir_re=re.compile("^[0-9]{6}_[A-Z0-9\-]+_[0-9]{4}_[A-Z0-9\-]{10,16}$")
+    for data_dir in CONFIG['bioinfo_tab']['data_dirs']:
+        if os.path.exists(data_dir):
+            potential_run_dirs=glob.glob(os.path.join(data_dir, '*'))
+            for run_dir in potential_run_dirs:
+                if rundir_re.match(os.path.basename(os.path.abspath(run_dir))) and os.path.isdir(run_dir):
+                    logger.info("Working on {}".format(run_dir))        
+                    update_statusdb(run_dir)
+
+
+def update_statusdb(run_dir):
+    project_ids=get_ss_projects(run_dir)
+    run_name = os.path.basename(os.path.abspath(run_dir))
+    status=get_status(run_dir)
+    couch=setupServer(CONFIG)
+    db=couch['bioinfo_analysis']
+    view = db.view('full_doc/pj_run_to_doc')
+    for p in project_ids:
+        obj={'run_id':run_name, 'project_id':p, 'status':status}
+        if len(view[[p, run_name]].rows) == 1:
+            remote_doc= view[[p, run_name]].rows[0].value
+            remote_status=remote_doc["status"]
+            if remote_status in ['Incoming', 'Sequencing Done', 'Demultiplexing', 'Demultiplexed', 'Transferring']:
+                final_obj=merge(obj, remote_doc)
+                logger.info("saving {} {} as  {}".format(run_name, p, status))
+                db.save(final_obj)
+        else:
+            logger.info("saving {} {} as  {}".format(run_name, p, status))
+            db.save(obj)
+
+
+
+def get_status(run_dir):
+    status='Incoming'
```

Comment:
Because I don't know if this class works for Hiseq and Miseq. And since at the last dev's meeting, Jun said that PR does not work, I have little interest in getting into that.